s3fs/tests/test_s3fs.py

```diff
@@ -939,3 +939,9 @@ def test_tags(s3):
     s3.touch(fname)
     s3.put_tags(fname, tagset)
     assert s3.get_tags(fname) == tagset
+
+    # Ensure merge mode updates value of existing key and adds new one
+    new_tagset = {'tag2': 'updatedvalue2', 'tag3': 'value3'}
+    tagset.update(new_tagset)
+    s3.put_tags(fname, tagset, mode='m')
```

Comment:
Should be `s3.put_tags(fname, new_tagset, mode='m')`, else you are not really testing merge :)

Reply:
Sorry my code didn't make it very clear what I was doing there -- I'm updating tagset with the values in new_tagset and then just reusing tagset:
```
>>> tagset = {'tag1': 'value1', 'tag2': 'value2'}
>>> new_tagset = {'tag2': 'updatedvalue2', 'tag3': 'value3'}
>>> tagset.update(new_tagset)
>>> tagset
{'tag1': 'value1', 'tag2': 'updatedvalue2', 'tag3': 'value3'}
```

Reply:
Yes, I see that the values are correctly updated, but `s3.put_tags(fname, tagset, mode='m')` uses the complete merged set of tags, not only the modifications - so it's not relying on grabbing the existing tags for the object first, it would work with mode 'o' too.

Reply:
Ah, I see what you mean.